dubbo-common/src/main/java/org/apache/dubbo/common/utils/ReflectUtils.java

```diff
@@ -1096,4 +1098,25 @@ public static boolean isPublicInstanceField(Field field) {
 
         return properties;
     }
+
+    public static Type[] getReturnTypes(Method method) {
+        Class<?> returnType = method.getReturnType();
+        Type genericReturnType = method.getGenericReturnType();
+        if (Future.class.isAssignableFrom(returnType)) {
+            if (genericReturnType instanceof ParameterizedType) {

```

Comment:
https://stackoverflow.com/questions/19906614/how-to-get-generic-return-type-of-a-method-using-reflections

Do you have checked the Generic Type ?

Reply:
I add a UT for this logic. Please help to check if it meets the requirement